### PR TITLE
FocusUtil: fix getPreviousElement to include previous sibling elements correctly

### PR DESCRIPTION
**Pull request by jspurlin**
#### Pull request checklist

- [X] Include a change request file using `$ npm run change`

#### Description of changes
The getPreviousElement functionality missed elements if a potential childMatch was found, but it was not tabbable (or tabbable is false) and the elemeent that should be found is a previous sibiling of childMatch.

Here;s some example: 
This worked:
<Focus starts before the zone, and is being put into the focusZone via the logic in FocusTrapZone>
FocusZone
   OverflowSetItem
      button (tabindex=0)
   OverflowSetItem
      button (tabindex=-1)
   OverflowSetItem
      button (tabindex=-1)
It goes down to the last button. The currentElement is on the last OverflowSetItem and it finds the button when looking for childMatch. From there determines it is not focusable (because of the isElementTabbable(childMatch, true /*checkTabIndex*/) call) so it would go to currentElement's previousSibling to continue looking (until it gets to the first button and then it finds athat as a match).

This did not work:
<Focus starts before the zone, and is being put into the focusZone via the logic in FocusTrapZone>
FocusZone
   OverflowSetItem
      button (tabindex=0)
      button (tabindex=-1)
      button (tabindex=-1)

It goes down to the last button. The currentElement is on the last OverflowSetItem and it finds the button when looking for childMatch. From there determines it is not focusable (because of the isElementTabbable(childMatch, true /*checkTabIndex*/) call) so it would go to currentElement's previousSibling to continue looking. Note that this skips all the other sibling buttons.

This fix is when we find a potential childMatch which we determine we cannot focus, move to that element's previous child instead of moving on to the next sibling of the parent (currentElement)

#### Focus areas to test
Verified that focus correctly moves to the previous element in all tested cases (including the two above)

### CODE REVIEW

File: packages/utilities/src/focus.ts [hunk 1]
```diff
@@ -99,8 +99,25 @@ export function getPreviousElement(
       allowFocusRoot,
       tabbable);
 
-    if (childMatch && ((tabbable && (isElementTabbable(childMatch, true))) || !tabbable)) {
-      return childMatch;
+    if (childMatch) {
```
Comment (dzearing) [inline packages/utilities/src/focus.ts]: Is it possible to write a unit test for this scenario?

Also, is there logic missing on "findNextElement"?

Comment (jspurlin) [reply packages/utilities/src/focus.ts]: getNextElement doesn't need to be updated, the logic is slightly different and doesn't check for tabindex when checking if an element is tabbable. There may be some other bug hidden in here from that difference, but it doesn't need this update

Comment (jspurlin) [reply packages/utilities/src/focus.ts]: and yes, I'll look at adding a unit test

Comment (dzearing) [reply packages/utilities/src/focus.ts]: cool ping me once you have a test to cover it (so we don't break it later!) and let's get it merged.

Comment (jspurlin) [reply packages/utilities/src/focus.ts]: @dzearing done, ut added. I also verified that this breaks without my fix and works with it :)

**Pull request merged**